connectivity/mqtt/duplex_callback.c

```diff
@@ -29,17 +31,65 @@ int mqtt_callback_logger_release() {
 }
 
 static status_t mqtt_request_handler(mosq_config_t *cfg, char *sub_topic, char *req) {
+  if (cfg == NULL || sub_topic == NULL || req == NULL) {
+    return SC_MQTT_NULL;
+  }
+
   status_t ret = SC_OK;
-  // TODO: process MQTT requests here. Deserialize the request and process it against corresponding api_* functions
+  char *json_result = NULL;
+  char device_id[ID_LEN];
+
+  // get the Device ID.
+  ret = mqtt_device_id_deserialize(req, device_id);
+  if (ret != SC_OK) {
+    log_error(mqtt_callback_logger_id, "[%s:%d:%d]\n", __func__, __LINE__, ret);
+    goto done;
+  }
 
-  // after finishing processing the request, set the response message with the following functions
+  if (strstr(sub_topic, "address")) {
+    ret = api_generate_address(&ta_core.tangle, &ta_core.service, &json_result);
+  } else if (strstr(sub_topic, "tag/hashes")) {
+    char tag[NUM_TRYTES_TAG];
```

Comment:
Can you split the following statements into 3 conditions? `tag`, `transaction`, and `tips`.

Reply:
But I think the current way is mre clear, like what we did in 
https://github.com/DLTcollab/tangle-accelerator/blob/develop/accelerator/http.c#L207

Or you mean the serialization only?

Reply:
Repeated `strstr` invocations might be inefficient and not readable.

Reply:
OK!